src/lib/quests/baseQuest.ts

```diff
@@ -0,0 +1,40 @@
+import MatchStatistics, { isPlayerInMatchFilter } from "../matchStatistics";
+import { type Quest, type QuestStatus, type QuestType } from "../quest";
+
+export abstract class BaseQuest<TConditionData, TState>
+  implements Quest<TConditionData, TState>
+{
+  abstract state: TState;
+  abstract type: QuestType;
+
+  constructor(
+    public conditionData: TConditionData,
+    public playerId: string,
+    public createdAt: Date,
+    public description: string,
+  ) {}
+
+  abstract evaluate(match: MatchWithPlayers): QuestStatus;
+
+  protected matchIsValidForQuest(match: MatchWithPlayers): boolean {
```

Comment:
Det her er måske lidt en farlig navngivning da den nogle gange er forkert el. diskuterbar

fx:

- Du skal vinde over en bestemt modstander og de fleste modstandere er derfor ikke gyldige
- Du skal splille en 1v1 så 2v2 er ikke gyldige
- Du skal vinde med en bestem point difference, så matches med færre point er ikke gyldige

Synes lidt den her funktion skal fjernes og hver enkel quest skal selv kalde de 2 metoder denne funktion kalder

Reply:
Der er et eller andet hvor jeg synes det er træls at skrive `!this.isPlayerInMatch(match) && !this.isMatchAfterQuestCreation(match)` i alle quests. Måske et andet navn ala `PlayerIsInMatchAndMatchDateValid` er nok ?